processor/src/decoder/mod.rs

```diff
@@ -124,13 +126,15 @@ impl Process {
             .ok_or(ExecutionError::MastNodeNotFoundInForest { node_id: node.on_false() })?
             .digest()
             .into();
-        let addr = self.chiplets.hash_control_block(
+        let (addr, result) = self.chiplets.hasher.hash_control_block(
```

Comment:
nit: can we rename `result` to `hashed_block`?

Same comment applies to all the other ones in this file